src/main/java/org/jenkinsci/plugins/workflow/steps/ArtifactArchiverStepExecution.java

```diff
@@ -32,6 +32,10 @@
     protected Void run() throws Exception {
         FilePath ws = getContext().get(FilePath.class);
         ws.mkdirs();
+        TaskListener listener = getContext().get(TaskListener.class);
+        if (listener != null) {
+            listener.getLogger().println("The 'archive' step is deprecated and may not work properly, please use 'archiveArtifacts' instead.");
```

Comment:
Well it works “properly” AFAIK, it is just unnecessary when you are on 2.x…so why not just bump up the Jenkins baseline to 2.7.3 like in other plugins, at which point you can simplify a whole bunch of test cases in this plugin which had been switching modes based on `Class.hasAnnotation`?